tensorpipe/transport/shm/connection.cc

```diff
@@ -337,7 +337,11 @@ void Connection::ReadOperation::handleRead(TConsumer& inbox) {
     const auto tup = inbox.readInTxWithSize<uint32_t>();
     const auto ret = std::get<0>(tup);
     const auto ptr = std::get<1>(tup);
-    TP_THROW_SYSTEM_IF(ret < 0, -ret);
+    if (ret < 0) {
```

Comment:
There are different errors that can happen. Which one should trigger the EOF error?

Reply:
In this case we should probably only trigger `EOFError` when `-ret == ENODATA`. We could also trigger `ShortRead` when read is larger than buffer size, and throw in other cases.